libindy_vdr/src/ffi/pool.rs

```diff
@@ -102,7 +110,9 @@ fn handle_pool_refresh(
             cache.update(&init_txns, latest_txns)?;
         }
         if let Some(new_txns) = new_txns {
-            let runner = PoolBuilder::new(config, new_txns).node_weights(node_weights).refreshed(true).into_runner()?;
+            // set this cache with unique key prefix
+            let txn_cache = read_lock!(LEDGER_CACHE_STRATEGY)?.as_ref().map(|s| Cache::new(s.clone(), init_txns.root_hash_base58().ok()));
```

Comment:
I believe it should be using the hash of the new transactions here, not the initial ones

Reply:
That would change the ID on a pool refresh though wouldn't it? Unless that's what we want. I went with the init_txns assuming those were the genesis transactions and were less likely to change. Additionally the pool cache seems to be using them already as a key so I figured we could use them as a (somewhat) stable key prefix for the txn cache

Reply:
I changed it to use the new_txns anyway but I figured I'd ask

Reply:
ACA-Py does its own caching, so in that case the genesis transactions will change over time anyway. If there are two successful refreshes, then it would also have different 'initial' transactions. I think for consistency and avoiding these edge cases we can just stick with the latest.